### PR TITLE
[nps] Added example with fixedwing

### PR DESCRIPTION
**Pull request by podhrmic**
Added NPS target for Bixler as an example

### CODE REVIEW

File: conf/airframes/examples/bixler_lisa_m_2.xml [hunk 1]
```diff
@@ -81,6 +85,33 @@
     <define name="BODY_TO_IMU_PHI"   value="0." unit="deg"/>
     <define name="BODY_TO_IMU_THETA" value="0." unit="deg"/>
     <define name="BODY_TO_IMU_PSI"   value="0." unit="deg"/>
+    
+    <!-- Dummy Gyro values for NPS -
```
Comment (flixr) [inline conf/airframes/examples/bixler_lisa_m_2.xml]: Doesn't really make sense to put "dummy" values for GYRO and ACCEL here... just omit them and the defaults for the IMU are used and you should be actually able to fly with them...

File: conf/airframes/examples/bixler_lisa_m_2.xml [hunk 2]
```diff
@@ -81,6 +85,33 @@
...
+    <define name="ACCEL_Y_SIGN" value="-1"/>
+    <define name="ACCEL_Z_SIGN" value="-1"/>
+    <define name="ACCEL_X_NEUTRAL" value="0"/>
+    <define name="ACCEL_Y_NEUTRAL" value="0"/>
+    <define name="ACCEL_Z_NEUTRAL" value="0"/>
+
+    <!-- Dummy Mag values for NPS -
+    replace with your own calibration -->
+    <define name="MAG_X_SIGN" value="1"/>
+    <define name="MAG_Y_SIGN" value="-1"/>
```
Comment (flixr) [inline conf/airframes/examples/bixler_lisa_m_2.xml]: any particular reason to change the signs here?

File: conf/airframes/examples/bixler_lisa_m_2.xml (hunk 1 above)
Comment (podhrmic) [inline conf/airframes/examples/bixler_lisa_m_2.xml]: It doesn't compile without the "dummy" values, I get this error:
```
In file included from /home/fwmav/Paparazzi/master/latest/paparazzi/sw/simulator/nps/nps_sensor_mag.h:4:0,
                 from /home/fwmav/Paparazzi/master/latest/paparazzi/sw/simulator/nps/nps_sensor_mag.c:1:
/home/fwmav/Paparazzi/master/latest/paparazzi/sw/simulator/nps/nps_sensor_mag.c: In function 'nps_sensor_mag_init':
/home/fwmav/Paparazzi/master/latest/paparazzi/conf/simulator/nps/nps_sensors_params_invariant.h:108:28: error: 'IMU_MAG_X_NEUTRAL' undeclared (first use in this function)
 #define NPS_MAG_NEUTRAL_X  IMU_MAG_X_NEUTRAL
                            ^
/home/fwmav/Paparazzi/master/latest/paparazzi/sw/airborne/math/pprz_algebra.h:125:15: note: in definition of macro 'VECT3_ASSIGN'
     (_a).x = (_x);        \
               ^
/home/fwmav/Paparazzi/master/latest/paparazzi/sw/simulator/nps/nps_sensor_mag.c:16:16: note: in expansion of macro 'NPS_MAG_NEUTRAL_X'
                NPS_MAG_NEUTRAL_X, NPS_MAG_NEUTRAL_Y, NPS_MAG_NEUTRAL_Z);
                ^
/home/fwmav/Paparazzi/master/latest/paparazzi/conf/simulator/nps/nps_sensors_params_invariant.h:108:28: note: each undeclared identifier is reported only once for each function it appears in
 #define NPS_MAG_NEUTRAL_X  IMU_MAG_X_NEUTRAL
                            ^
/home/fwmav/Paparazzi/master/latest/paparazzi/sw/airborne/math/pprz_algebra.h:125:15: note: in definition of macro 'VECT3_ASSIGN'
     (_a).x = (_x);        \
               ^
/home/fwmav/Paparazzi/master/latest/paparazzi/sw/simulator/nps/nps_sensor_mag.c:16:16: note: in expansion of macro 'NPS_MAG_NEUTRAL_X'
                NPS_MAG_NEUTRAL_X, NPS_MAG_NEUTRAL_Y, NPS_MAG_NEUTRAL_Z);
                ^
/home/fwmav/Paparazzi/master/latest/paparazzi/conf/simulator/nps/nps_sensors_params_invariant.h:109:28: error: 'IMU_MAG_Y_NEUTRAL' undeclared (first use in this function)
 #define NPS_MAG_NEUTRAL_Y  IMU_MAG_Y_NEUTRAL
                            ^
/home/fwmav/Paparazzi/master/latest/paparazzi/sw/airborne/math/pprz_algebra.h:126:15: note: in definition of macro 'VECT3_ASSIGN'
     (_a).y = (_y);        \
               ^
/home/fwmav/Paparazzi/master/latest/paparazzi/sw/simulator/nps/nps_sensor_mag.c:16:35: note: in expansion of macro 'NPS_MAG_NEUTRAL_Y'
                NPS_MAG_NEUTRAL_X, NPS_MAG_NEUTRAL_Y, NPS_MAG_NEUTRAL_Z);
                                   ^
/home/fwmav/Paparazzi/master/latest/paparazzi/conf/simulator/nps/nps_sensors_params_invariant.h:110:28: error: 'IMU_MAG_Z_NEUTRAL' undeclared (first use in this function)
 #define NPS_MAG_NEUTRAL_Z  IMU_MAG_Z_NEUTRAL
                            ^
/home/fwmav/Paparazzi/master/latest/paparazzi/sw/airborne/math/pprz_algebra.h:127:15: note: in definition of macro 'VECT3_ASSIGN'
     (_a).z = (_z);        \
               ^
/home/fwmav/Paparazzi/master/latest/paparazzi/sw/simulator/nps/nps_sensor_mag.c:16:54: note: in expansion of macro 'NPS_MAG_NEUTRAL_Z'
                NPS_MAG_NEUTRAL_X, NPS_MAG_NEUTRAL_Y, NPS_MAG_NEUTRAL_Z);
                                                      ^
make[1]: *** [/home/fwmav/Paparazzi/master/latest/paparazzi/var/aircrafts/Bixler/nps/nps/nps_sensor_mag.o] Error 1
make[1]: *** Waiting for unfinished jobs....
In file included from /home/fwmav/Paparazzi/master/latest/paparazzi/sw/simulator/nps/nps_fdm_jsbsim.cpp:49:0:
/usr/include/JSBSim/models/propulsion/FGThruster.h:100:16: warning: unused parameter 'rpm' [-Wunused-parameter]
   virtual void SetRPM(double rpm) {};
                ^
/usr/include/JSBSim/models/propulsion/FGThruster.h:101:16: warning: unused parameter 'rpm' [-Wunused-parameter]
   virtual void SetEngineRPM(double rpm) {};
                ^
make[1]: Leaving directory `/home/fwmav/Paparazzi/master/latest/paparazzi/sw/airborne'
make: *** [nps.compile] Error 2
make: Leaving directory `/home/fwmav/Paparazzi/master/latest/paparazzi'
```

File: conf/airframes/examples/bixler_lisa_m_2.xml (hunk 2 above)
Comment (podhrmic) [inline conf/airframes/examples/bixler_lisa_m_2.xml]: this is what I got from another airframe where those were defined. Apparently the signs are not needed, so I ll remove them.

**Comment by flixr** [comment]
You need a mag calibration part, as we can't really provide defaults for that...
But for accel and gyro the defaults (from datasheet) should be good enough in most cases...

**Comment by podhrmic** [comment]
OK, this should fix it.

**Pull request closed without merging**